test/crud.test.js

```diff
@@ -0,0 +1,170 @@
+const rider = {
+  _id: 'rider1',
+  creatorId: 'c724614a-3c9b-42de-8db6-175224b9bbb8',
+  createdAt: '2019-04-16T10: 15: 51.303Z',
+  updaterId: 'c724614a-3c9b-42de-8db6-175224b9bbb8',
+  updatedAt: '2019-04-16T10: 16: 01.076Z',
+  __STATE__: 'PUBLIC',
+  name: 'Mario',
+  surname: 'Rossi',
+  date: '2019-04-01T22: 00: 00.000Z',
+  position: [
+    9.18380000000002,
+    45.45391,
+  ],
+  email: 'user1@fake-domain.com',
+  skypeId: 'user1',
+  chat: 'user1',
+}
+
+async function setupFastify(envVariables) {
+  const fastify = await lc39('./index.js', {
+    logLevel: 'silent',
+    envVariables,
+  })
+  return fastify
+}
+
+tap.test('Testing CRUD', async test => {
+  const fastify = await setupFastify({
+    USERID_HEADER_KEY: 'userid',
+    GROUPS_HEADER_KEY: 'groups',
+    CLIENTTYPE_HEADER_KEY: 'clienttype',
+    BACKOFFICE_HEADER_KEY: 'backoffice',
+    MICROSERVICE_GATEWAY_SERVICE_NAME: 'microservice-gateway.example.org',
+    CRUD_PATH: 'crud-service',
+  })
+  test.tearDown(() => fastify.close())
+
+  test.test('GET /riders/ route CORRECTLY', async assert => {
+    assert.plan(3)
+    const scope = nock('https://crud-service')
+      .get('/v2/riders')
+      .reply(200, riders)
+    const response = await fastify.inject({
+      method: 'GET',
+      url: '/riders',
+    })
+    const expectedResponse = {
+      statusCode: 200,
+      result: riders,
+    }
+    assert.strictSame(response.statusCode, 200)
+    assert.strictSame(JSON.parse(response.body), expectedResponse)
+    assert.ok(scope.isDone())
+  })
+  test.test('GET /riders/:id route CORRECTLY', async assert => {
+    assert.plan(3)
+
+    const scope = nock('https://crud-service')
+      .get('/v2/riders/rider1')
+      .reply(200, rider)
+
+    const response = await fastify.inject({
+      method: 'GET',
+      url: '/riders/rider1',
+    })
+
+    const expectedResponse = {
+      statusCode: 200,
+      result: rider,
+    }
+
+    assert.strictSame(response.statusCode, 200)
+    assert.strictSame(JSON.parse(response.payload), expectedResponse)
+    assert.ok(scope.isDone())
+  })
+
+  test.test('GET /riders/ route SERVER ERROR', async assert => {
+    assert.plan(3)
+
+    const scope = nock('https://crud-service')
+      .get('/v2/riders')
+      .reply(500, 'Internal Server Error')
```

Comment:
Crud service returns an object with `statusCode`, `error` (_Internal Server Error_) and `message` (some custom message); change all replies in these tests accordingly